### ISSUE TITLE
[Suggestion] To be able to disable a command line

### ISSUE THREAD

**Issue by leobenkel** [scallop/scallop#181]
Hey,
In a previous project, long time ago. I designed a command line parser library.

One feature that I had, which I thought was great, was to be able to turn off a command line.

Let's say you have this big jar, with a lot of arguments:

```
....jar --something value --foo bar ...
```

If you want to remove one parameter, you will need to erase the `--foo bar` and it can sometime be a pain if the value is long, of a path somewhere and you dont want to lose it just because you are testing something.

What I implemented what that you can use `//` instead of `--` and the parser will parse the command but not save the result. It will act as if it was not there and ignore all the values associated with it as well.

So you would have 
```
...jar --something value //foo bar --somethingelse value
```
And `//foo bar` will just be ignored. 

Maybe we could add that to Scallop.

Thanks !

**Comment by Rogach** [comment]
Hi! I think this can be done using standard shell (or other programming language) options.

For example in bash you can use `#` to comment out the unused arguments. And the commented arguments will be still saved in history, so they will remain easily accessible.

Thus I don't think it will be useful to implement such feature directly in Scallop - it will only complicate maintenance and introduce possible bugs.

**Comment by leobenkel** [comment]
@Rogach , `bash` isn't aware of `--foo 1 2 3` with dependencies between the blocks. if you do `# foo 1 2 3` , `1 2 3` will still be passed to the application.

**Comment by Rogach** [comment]
@leobenkel Are you sure? Here's what I tried:

```bash
$ echo bar # --foo 1 2 3
bar
```

Or do you mean that you actually want to the remainders to be passed? For example:

```
invocation : program --something value //foo bar --somethingelse value
args parsed: program --something value --somethingelse value
```

**Comment by leobenkel** [comment]
yes i meant the second option. When you have a long series of argument, and you are doing some testing. You might want to remove one argument for this one test. then put it back. Without changing anything else.

**Comment by Rogach** [comment]
Ok. I'm a bit conflicted - the feature looks nice, but enabling it for everybody may break something. On the other hand, if we play it safe and hide it behind a feature flag then nobody will use it and it will get no traction.

At the very least I'm sure we should only react to `//` as a separate argument, not as a prefix (`//foo`), because some real-world arguments can be of that form (xpath comes to mind, I'm sure there are more examples).

If this separator appears in the argument list, then we disable only the rest of the argument list: `--opt a // b c` => `--opt a`

If this separator immediately precedes the option, then we disable the option and its arguments: `--opt1 // --opt2 a b --opt3` => `--opt1 --opt3`

And we need to provide some way of escaping this, to make it possible to pass `//` argument explicitly if the need arises. I have no ideas here - maybe you have a suggestion?

Again, I was confused at first and closed the issue, but now it looks like a reasonable functionality. However, I'm not sure that I will be able to find the time and energy in the near future to implement it myself - PR will be welcome. (here's the main parse method, most probably the meat of this functionality should be implemented there: [Scallop.parse](https://github.com/scallop/scallop/blob/9693fd07d9e976482f57721c1db006f165d133f8/src/main/scala/org.rogach.scallop/Scallop.scala#L104). Caveat emptor: this method is one of the oldest in Scallop, it was first written 8 (or 9?) years ago and some refactoring is also quite welcome)

**Comment by leobenkel** [comment]
@Rogach thank you for taking the time to understand what I meant. 

I like the idea of having `//` as a "magic argument".

I was suggesting to use `//foo` because I expected the `--` block parser to already be parsed and adding a parser to find `(--|//)` instead with a boolean `isActive` sounded easier to me. 

But I like the mechanic you are suggesting better. Even for the use case I described where you want to test something it is easier to add or remove a single `//` rather than replacing `--` by `//`. 

I am quite busy as well at the moment.
I am in the middle of founding my own company _but it's a secret for the moment_ :) 

I will give it a shot as soon as I have time available tho ! 

Thank you again for your time.
